packages/nodes-base/nodes/Gitlab/Gitlab.node.ts

```diff
@@ -1229,12 +1544,134 @@ export class Gitlab implements INodeType {
 
 						endpoint = `/users/${owner}/projects`;
 					}
+				} else if (resource === 'file') {
+					if (['create', 'edit'].includes(operation)) {
+						// ----------------------------------
+						//         create
+						// ----------------------------------
+
+						requestMethod = operation === 'create' ? 'POST' : 'PUT';
+
+						const filePath = this.getNodeParameter('filePath', i);
+						const additionalParameters = this.getNodeParameter(
+							'additionalParameters',
+							i,
+						) as IDataObject;
+
+						body.branch = this.getNodeParameter('branch', i) as string;
+						body.commit_message = this.getNodeParameter('commitMessage', i) as string;
+
+						if (additionalParameters.author) {
+							body.author = additionalParameters.author;
+						}
+						if (
+							additionalParameters.branchStart &&
+							(additionalParameters.branchStart as IDataObject).branchStart
+						) {
+							body.start_branch = (additionalParameters.branchStart as IDataObject).branchStart;
+						}
+
+						if (this.getNodeParameter('binaryData', i)) {
+							// Is binary file to upload
+							const item = items[i];
+
+							if (item.binary === undefined) {
+								throw new NodeOperationError(this.getNode(), 'No binary data exists on item!', {
+									itemIndex: i,
+								});
+							}
+
+							const binaryPropertyName = this.getNodeParameter('binaryPropertyName', i);
+
+							if (item.binary[binaryPropertyName] === undefined) {
+								throw new NodeOperationError(
+									this.getNode(),
+									`No binary data property "${binaryPropertyName}" does not exists on item!`,
+									{ itemIndex: i },
+								);
+							}
+
+							// Currently internally n8n uses base64 and also Github expects it base64 encoded.
```

Comment:
Might be worth changing this from Github to Gitlab

Reply:
Done.